drivers/bluetooth/hci/hci_spi_st.c

```diff
@@ -435,12 +435,12 @@ static int bt_spi_rx_buf_construct(uint8_t *msg, struct net_buf **bufp, uint16_t
 			len = sizeof(iso_hdr) + bt_iso_hdr_len(sys_le16_to_cpu(iso_hdr.len));
 		} else {
 			LOG_ERR("No available ISO buffers!");
-			return NULL;
+			return -ENOMEM;
```

Comment:
Would
```suggestion
			return -ENOBUFS;
```
have been more appropriate?

Reply:
Either way is fine with me, although it would be good to be consistent with the choice of error number for all similar situations throughout the Bluetooth stack.

Reply:
Looking at the code, there are two kind of errors related to buffer:
1. There are no buffers
2. Buffer is available, but it doesn’t have enough space for the data.

Is it appropriate to have ENOBUFS for case 1 and ENOMEM for case 2 ?

We could file an issue to keep track of this problem and address it in a future or. What do you think?

As a side note, when I read the explanation for error ENOMEM, I see “Not enough core”, which means nothing to me. Is it correct?

Reply:
> Is it appropriate to have ENOBUFS for case 1 and ENOMEM for case 2 ?

Yeah, but that can be done later if we really want. 

> As a side note, when I read the explanation for error ENOMEM, I see “Not enough core”, which means nothing to me. Is it correct?

That depends on the libc implementation I guess. I see that both armstdc and minimal in Zephyr defines it as such, but I agree it sounds odd. On my system and from https://www.man7.org/linux/man-pages/man3/errno.3.html it's defined as `ENOMEM Not enough space/cannot allocate memory (POSIX.1-2001).` which makes more sense IMO